R/import_list.R

```diff
@@ -47,8 +47,12 @@ function(file,
     if (missing(setclass)) {
         setclass <- NULL
     }
+    strip_exts <- function(file) {
+      exts <- paste0("\\.", lapply(file, tools::file_ext), "$")
+      sapply(seq_along(file), function(x) gsub(exts[x], "", file[x]))
+    }
     if (length(file) > 1) {
-        names(file) <- gsub(paste0("\\.", tools::file_ext(file[1]), "$"), "", file, ignore.case = TRUE)
+        names(file) <- strip_exts(file)
```

Comment:
See https://github.com/leeper/rio/pull/164#discussion_r144111269

Reply:
There's a function `tools::file_path_sans_ext` that might be useful (possibly in combination with `basename()`

Reply:
Great call, thanks @HughParsonage.

@leeper ready for merge, I think